chapters/part3/fairness.md

```diff
@@ -722,9 +722,289 @@ We have made the [bias-mitigated, finetuned large RoBERTA model for SNLI](https:
+
+After 10 epochs of finetuning:
+```python
+"adversarial_bias_mitigated_predictor_metrics": {
+    "net_neutral": 0.613096454815352,
+    "fraction_neutral": 0.6704967487937075,
+    "threshold_0.5": 0.6637061892722586,
+    "threshold_0.7": 0.49490217463150243
+},
+"baseline_predictor_metrics": {
+    "net_neutral": 0.49562667062645066,
+    "fraction_neutral": 0.506870600337101,
+    "threshold_0.5": 0.47600531264458984,
+    "threshold_0.7": 0.30368001850750215
+}
+```
+
+The `NaturalLanguageInference` scores for the adversarial-bias-mitigated, 10-epoch-finetuned predictor are much closer to 1 than they are for the 1-epoch-finetuned predictor and the baseline, pretrained predictor. This suggests lower binary gender bias with respect to occupations, as bias will result in a higher probability of entailment or contradiction. However, the bias-mitigated, finetuned model from the previous section boasts slightly higher `NaturalLanguageInference` scores for this downstream task.
+
+Unlike for the bias-mitigated model from the previous section, the adversarial-bias-mitigated predictor with the highest validation accuracy (in this case, 1 epoch of finetuning) does not demonstrate a significant reduction in binary gender bias, and the adversarial-bias-mitigated predictor seems to benefit from more epochs of finetuning. 
+
+`diff.txt` allows us to see specific examples for which the adversarial-bias-mitigated, 10-epoch-finetuned predictor chose neutral but the baseline, pretrained model didn't:
+
+```python
+{
+    "sentence_pair": "<s>A zoologist spoke to an adult.</s></s>A woman spoke to an adult.</s><pad><pad><pad><pad>",
+    "adversarial_bias_mitigated_label": "neutral",
+    "baseline_label": "entailment"
+}
+```
+
+We have made the [adversarial-bias-mitigated, finetuned large RoBERTA model for SNLI](https://storage.googleapis.com/allennlp-public-models/adversarial-binary-gender-bias-mitigated-snli-roberta.2021-06-17.tar.gz) publicly available with a [model card](https://github.com/allenai/allennlp-models/blob/main/allennlp_models/modelcards/pair-classification-adversarial-binary-gender-bias-mitigated-roberta-snli.json). You can play around with the model on the [AllenNLP demo](https://demo.allennlp.org/textual-entailment/adv-bin-gen-bias-mitigated-roberta-snli).
+
+## Pitfalls of Adversarial Bias Mitigation
+
+Adversarial bias mitigation does have some pitfalls. For one, both the predictor and adversary are black boxes, which makes it uninterpretable in contrast to the bias mitigators described in previous sections. Furthermore, while [1] derives theoretical guarantees for adversarial bias mitigation, the proof relies on the strong, and impractical, assumption that the loss functions of the predictor and adversary are convex with respect to their parameters. Finally, achieving convergence in adversarial networks is tricky in practice, especially for large, contextual language models; some tips for training adversarial networks include to [2]:
```

Comment:
```suggestion
Adversarial bias mitigation does have some pitfalls. For one, both the predictor and adversary are black boxes, which makes it uninterpretable in contrast to the bias mitigators described in previous sections. Furthermore, while [1] derives theoretical guarantees for adversarial bias mitigation, the proof relies on the strong, and impractical, assumption that the loss functions of the predictor and adversary are convex with respect to their parameters. Finally, achieving convergence in adversarial networks is tricky in practice, especially for large, contextual language models; **some tips for training adversarial networks** include to [2]:
```

Reply:
Note: the suggestion is to **bold** the tips part; I don't think that's visible in github edits.